### DISCUSSION TITLE
Will ServiceMonitor scrape non-ready targets?

### DISCUSSION THREAD

**Discussion by baryluk** [prometheus-operator/prometheus-operator#5693] [Q&A]
It is unclear from documentation.

AFAIK Service only routes to endpoints that are for pods (or containers?) that are in ready state.

But that is now what one usually wants for monitoring. For monitoring, one wants to monitor and scrape even the non-ready ones.

**Comment by simonpasquier** [comment]
You probably want to use PodMonitor and set `filterRunning: false`. See https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMetricsEndpoint

**Reply by baryluk** [reply]
I definitively do not want that. If they are not running (Failed, or Succeeded), there is nothing to scrape.

But, pods that are not ready, are definitively is something I do want to scrape.

**Comment by simonpasquier** [comment]
By definition, unready pods will not be included in the Kubernetes endpoints backing the service so you want PodMonitor anyway.
